waku/v2/protocol/waku_swap/waku_swap.nim

```diff
@@ -207,25 +207,27 @@ proc init*(wakuSwap: WakuSwap) =
     info "received cheque", value=res.value
     wakuSwap.handleCheque(res.value)
 
-  proc credit(peerId: PeerId, n: int) {.gcsafe, closure.} =
+  proc credit(peerId: PeerId, n: int) {.closure.} =
```

Comment:
Did you look at how other places with metrics handle this? Not sure if it can be done gcsafe or not.